src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -359,27 +434,34 @@ function calc_volume_integral!(du, u, volume_integral, mesh,
   @unpack entropy_projected_u_values, Ph, sparsity_pattern = cache
   @unpack fluxdiff_local_threaded, rhs_local_threaded = cache
   rd = dg.basis
-  skip_index(i,j) = i > rd.Nq && j > rd.Nq
 
   flux_conservative, flux_nonconservative = volume_integral.volume_flux
 
   @threaded for e in eachelement(mesh, dg, cache)
     fluxdiff_local = fluxdiff_local_threaded[Threads.threadid()]
     fill!(fluxdiff_local, zero(eltype(fluxdiff_local)))
     u_local = view(entropy_projected_u_values, :, e)
-    for i in eachdim(mesh)
-      # Todo: DGMulti. Dispatch on curved meshes, this code only works for affine meshes (accessing rxJ[1,e],...)
-      Qi_skew = build_lazy_physical_derivative(e, i, mesh, dg, cache)
 
-      # compute conservative flux differencing contribution
-      hadamard_sum!(fluxdiff_local, Qi_skew, flux_conservative, i,
-                    u_local, equations, sparsity_pattern, skip_index)
+    # Val{true}() indicates flux is symmetric
+    local_flux_differencing!(fluxdiff_local, u_local, e,
+                             Val{true}(), flux_conservative,
+                             has_sparse_operators(dg),
+                             mesh, equations, dg, cache)
+
+    # Compute non-conservative contributions. Non-conservative fluxes for MHD don't support
+    # `orientation::AbstractVector`, so we use numerical fluxes in Cartesian directions
+    # and sum up the discrete derivative operators per coordinate direction accordingly.
```

Comment:
They do support arbitrary directions. However, you need to use the signature
```julia
    flux_nonconservative_powell(u_ll, u_rr,
                                normal_direction_ll     ::AbstractVector,
                                normal_direction_average::AbstractVector,
                                equations::IdealGlmMhdEquations2D)
```
on general curved elements. This is more complex since we need both the `normal_direction` at the first argument position and also the average of both `normal_directions`. In your case, you don't have curvilinear coordinates right now. Thus, the contravariant vectors are constant in each element. Hence, you could call the nonconservative flux with two identical `normal_direction`s. 

Alternatively, we could implement the fallback `flux_nonconservative_powell(u_ll, u_rr, normal_direction::AbstractVector, equations::IdealGlmMhdEquations2D) = flux_nonconservative_powell(u_ll, u_rr, normal_direction, normal_direction, equations)`. However, if we had done that before, you would have used it directly. This would have been fine here but you would have introduced some bugs if you used it with curved elements. Hence, I'm not sure whether we want to open the door for hard-to-find bugs here.... That's a difficult question. What do you think, @jlchan?